File: nonsquares.rb
#!/usr/bin/env ruby
# http://rosettacode.org/wiki/Sequence_of_non-squares

def nonsquare(number)
  (number + ((0.5) + Math.sqrt(number))).floor
end  	

def show_sequence_range(start_number, end_number)
  seq = {}
  (start_number..end_number).each { |n| seq[n] = nonsquare(n) }
  return seq
end

def test_below(limit)
  (1...limit).each { |i| return false if Math.sqrt(nonsquare(i)) % 1 == 0 }
  return true
end		

p show_sequence_range(1, 22)	
p test_below(1_000_000)

File: nonsquares.swift
// http://rosettacode.org/wiki/Sequence_of_non-squares
import Darwin

func nonSquares(n:Int) -> Float {
  return Float(n) + floor(Float(1 / 2 + sqrt(Float(n))))
}

for var i = 1; i <= 22; i++ {
  let nonSquare = Int(nonSquares(i))
  println("\(i): \(nonSquare)")
}

for var j = 23; j <= 1_000_000; j++ {
  let nonSquare = nonSquares(j)
  if sqrt(nonSquare) % 1 == 0 {
    println("\(j) is a square.")
  }
}

println("I didn't find any squares for n <= 1_000_000.")
